src/Path.php

```diff
@@ -189,7 +189,7 @@ public static function check($path)
 	 */
 	public static function clean($path, $ds = DIRECTORY_SEPARATOR)
 	{
-		if (!is_string($path))
+		if (!is_string($path) && !empty($path))
```

Comment:
If it's not a string and not empty, throw an exception?  Maybe I'm not thinking straight but something seems funny about that.